src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -0,0 +1,278 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use super::syscall_request::*;
+use crate::business_logic::execution::objects::*;
+use crate::core::errors::syscall_handler_errors::SyscallHandlerError;
+use crate::core::syscalls::syscall_handler::SyscallHandler;
+use cairo_rs::types::relocatable::{MaybeRelocatable, Relocatable};
+use cairo_rs::vm::vm_core::VirtualMachine;
+use num_bigint::BigInt;
+use num_traits::ToPrimitive;
+
+//* -----------------------------------
+//* BusinessLogicHandler implementation
+//* -----------------------------------
+
+pub struct BusinessLogicSyscallHandler {
+    tx_execution_context: Rc<RefCell<TransactionExecutionContext>>,
+    events: Rc<RefCell<Vec<OrderedEvent>>>,
+}
+
+impl BusinessLogicSyscallHandler {
+    pub fn new() -> Result<Self, SyscallHandlerError> {
+        let events = Rc::new(RefCell::new(Vec::new()));
+        let tx_execution_context = Rc::new(RefCell::new(TransactionExecutionContext::new()));
+        Ok(BusinessLogicSyscallHandler {
+            events,
+            tx_execution_context,
+        })
+    }
+}
+
+impl SyscallHandler for BusinessLogicSyscallHandler {
+    // trait functions
+    fn emit_event(
+        &self,
+        vm: &VirtualMachine,
+        syscall_ptr: Relocatable,
+    ) -> Result<(), SyscallHandlerError> {
+        let SyscallRequest::EmitEvent(request) =
+            self._read_and_validate_syscall_request("emit_event", vm, syscall_ptr)?;
+
+        println!("syscall request: {:?}", request);
+        let keys_len = bigint_to_usize(&request.keys_len)?;
+        let data_len = bigint_to_usize(&request.data_len)?;
```

Comment:
Why are these `BigInt` to begin with?

Reply:
thus is how they are returned by the memory in cairo-rs